### PR TITLE
[FIX] stock_by_warehouse: Quantity Available

### PR DESCRIPTION
**Pull request by CarmenMiranda**
Available quantities were being retrieved not only for the exact location but also for their child locations. That caused some product
to be double-counted when computing quantities.

This commit fixes the above.

### CODE REVIEW

File: stock_by_warehouse/models/product_product.py [hunk 1]
```diff
@@ -43,38 +43,20 @@ def _prepare_domain_available_not_reserved(self):
...
 
-            vals['qty_available_not_res'] = quantity
-            res[prod.id] = vals
-        self._product_available_not_res_hook(quants)
-
-        return res
+        res = {}
+        for quant in quants:
+            quantity = quant.get('quantity') - quant.get('reserved_quantity')
+            res[quant.get('product_id')[0]] = quantity
```
Comment (luisg123v) [inline stock_by_warehouse/models/product_product.py]: Could you add the quantity only if it's positive

Comment (CarmenMiranda) [reply stock_by_warehouse/models/product_product.py]: `res[quant.get('product_id')[0]] = quantity > 0 and quantity or 0.0`

or

```
if quantity > 0:
    res[quant.get('product_id')[0]] = quantity
```

Comment (luisg123v) [reply stock_by_warehouse/models/product_product.py]: The first one but shorter:

`res[quant.get('product_id')[0]] = max(quantity, 0.0)`

Comment (CarmenMiranda) [reply stock_by_warehouse/models/product_product.py]: Done :smile:

**Review by luisg123v — APPROVED** [review]

**Pull request merged**